Reorganize Food Recipes, "Balance" Tweaks

Owai-Seek opened the pull request:
-Reorganized -ALL- foods in the crafting menu.
-Made a section for Eastern Dishes, Mexican Dishes, and Sweets
-Made Bowls, Salads, Soups, Eggs Benedict, and Omelettes normal sized.
-Moved Easter Recipes to the food files, fixing it in the process.
-Capitalized a few things.

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Made subsections in recipes files to organize things based on type, as well as alphabetical sorting. Some minor spellchecks and "balance" changes.

Section Examples:
Bread, Toast, and Misc Bread.
Standard, Exotic, Mystical, and Colored burgers.
Fruit Cake, Fancy Cake, Special Cake
Sorted Frozen Section
Kebabs, Spider, Fish, and Misc (in Meat)
A new Mexican file for Burritos, Tacos, Enchiladas, ect.
A thinned out Misc section.
An only -slightly- organised Pastry section. (Donuts are now sorted.)
Fruit, Other Pies, and Tarts.
Alphabetized Pizza, as well as a separation of Arnold/Anomaly from regular pies.
Sorted Salads
Sandwiches -mostly- untouched.
Soups and Spaggetti Sorted
Sushi is now Eastern
New Sweets Category

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Foods are an absolute mess to find. There was a massive unorganized cluster of related foods in the misc category, there was toast in both breads /and/ toast, things werent put together in any cohesive way at all, and it made me irritated.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
tweak: Reorganized all food recipes, (hopefully) making things easier to find.
balance: Trays now have a whitelist, allowing them to pick up normal sized foods, bowls, glassware, booze, ect.
fix: Easter foods are no longer their own Misc Food Category on the top of the menu.
spellcheck: fixed a few typos/capitalization consistency.
tweak: Mexican Foods are their own Subcategory.
tweak: Donuts are their own Subcategory
tweak: Moved Sweets from Misc Food in with Pies. Renamed to Pies & Sweets
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

## (Original) Screenshots

![kwd1I2IQuS](https://user-images.githubusercontent.com/7543955/74657706-d32def00-515e-11ea-84b9-693f3de48594.png)
![TTMbVo9ag3](https://user-images.githubusercontent.com/7543955/74657709-d3c68580-515e-11ea-968c-6a8c68c30d39.png)
![UcXs8XB41j](https://user-images.githubusercontent.com/7543955/74657710-d45f1c00-515e-11ea-87c2-c5653e83e18a.png)
![CMDUhLjTfb](https://user-images.githubusercontent.com/7543955/74657719-d75a0c80-515e-11ea-9c94-f8c053f7e5d4.png)

## Category Updates

![fJbidtEPj9](https://user-images.githubusercontent.com/7543955/75043228-b7438980-548d-11ea-80ee-76a8f51f87d4.png)
![wMWuUF4XMp](https://user-images.githubusercontent.com/7543955/75043232-ba3e7a00-548d-11ea-94a6-f083db67cf51.png)
![MW6UZaLMnP](https://user-images.githubusercontent.com/7543955/75043234-bca0d400-548d-11ea-92e7-e6fbd599f432.png)
![x5oDdwZJ2Y](https://user-images.githubusercontent.com/7543955/75043239-bf032e00-548d-11ea-8ddf-c5a9688bd0e3.png)
![1MSyMsgvSu](https://user-images.githubusercontent.com/7543955/75043244-c1658800-548d-11ea-90e1-a4616f2934c3.png)
![image](https://user-images.githubusercontent.com/7543955/75044438-a3992280-548f-11ea-8dca-710704861677.png)

Owai-Seek committed:
Reorganise Food + Balance Tweaks

-Reorganised -ALL- foods in the crafting menu.
-Made a section for Eastern Dishes, Mexican Dishes, and Sweets
-Made Bowls, Salads, Soups, Eggs Benedict, and Omelettes normal sized.
-Moved Easter Recipies to the food files, fixing it in the process.
-Capitalised a few things.

Owai-Seek committed:
Sorted Frozen Foods

Owai-Seek committed:
Moving Burgers Around

Trilbyspaceclone commented on code/__DEFINES/construction.dm:
```diff
@@ -102,7 +104,7 @@
 #define CAT_SANDWICH	"Sandwiches"
 #define CAT_SOUP	"Soups"
 #define CAT_SPAGHETTI	"Spaghettis"
-#define CAT_FISH	"Fish"
```
Did you mean to replace Fish?

Owai-Seek committed:
Renamed CAT_FISH to CAT_EASTERN

Ghommie commented on code/__DEFINES/construction.dm:
```diff
@@ -102,7 +104,7 @@
 #define CAT_SOUP	"Soups"
 #define CAT_SPAGHETTI	"Spaghettis"
-#define CAT_FISH	"Fish"
+#define CAT_SWEETS	"Sweets"
```
Being a super-category of most cakes, pies and pastries, it is likely to cause some confusion (specially with the latter).
Just rename the "Pastries" category to "Confectioneries".

Ghommie commented on code/modules/food_and_drinks/food/customizables.dm:
```diff
@@ -294,7 +294,7 @@
 	reagent_flags = OPENCONTAINER
 	materials = list(MAT_GLASS = 500)
-	w_class = WEIGHT_CLASS_NORMAL
+	w_class = WEIGHT_CLASS_SMALL
```
A better long term solution would be adding a "size whitelist" variable to the storage component for typecaches of items that can be inserted regardless of size and then whitelist bowls and food reagent containers for the tray.

Ghommie commented on code/modules/food_and_drinks/food/snacks_other.dm:
```diff
@@ -674,3 +674,49 @@
+	name = "brioche cake slice"
+	desc = "Delicious sweet-bread. Who needs anything else?"
+	icon_state = "briochecake_slice"
+	filling_color = "#FFD700"
```
Being a cake subtype, this is better found in the cakes file.

Ghommie commented on code/modules/food_and_drinks/food/snacks_other.dm:
```diff
@@ -674,3 +674,49 @@
+	bonus_reagents = list(/datum/reagent/consumable/nutriment = 2, /datum/reagent/consumable/nutriment/vitamin = 2)
+	bitesize = 3
+	filling_color = "#FFFFF0"
+	list_reagents = list(/datum/reagent/consumable/nutriment = 6)
```
Egg snacks file.

Ghommie commented on code/modules/food_and_drinks/recipes/tablecraft/recipies_sweets.dm:
```diff
@@ -0,0 +1,117 @@
+// see code/module/crafting/table.dm
```
File name typo. Also this should be merged with the pastries file.

Ghommie reviewed (changes requested):
Most times, food recipes are better indexed by typology, main ingredients and food group rather than cultural origins to help people choose what they want to cook based on the available ingredients.

So no, I can't approve of the mexican and eastern categories and you should revert them.

Owai-Seek committed:
Resolving some convos.

Rename Pastries to Pastry & Sweets
Move all sweets to Pastry & Sweets
Revert Weight Class Changes
Add Whitelist to trays. Glasswear, Drinks, Foods, Kitchen Items, ect.
Move Brioche Cake item/recipe
Move Scotch Egg item/recipe

Owai-Seek committed:
I forgot to hit save.

Owai-Seek committed:
Fix, New Category, file renames.

Remove Sweets Category
Add Donut Category
Rename recipies_sushi.dm to recipies_eastern_fish.dm (for sorting in the UI)
Rename recipies_pie.dm to recipies_pie_sweets.dm
Remove Onigiri from Misc (It's in Eastern)

Ghommie commented on code/__DEFINES/construction.dm:
```diff
@@ -92,17 +92,19 @@
 #define CAT_BURGER	"Burgers"
 #define CAT_CAKE	"Cakes"
+#define CAT_DONUT	"Donuts"
+#define CAT_EASTERN	"Eastern & Fish"
```
I told you not to fill this category with miscellaneous eastern food. Bring those recipes where you found em.

Ghommie commented on code/modules/food_and_drinks/food/snacks_egg.dm:
```diff
@@ -124,7 +124,6 @@
 	bonus_reagents = list(/datum/reagent/consumable/nutriment = 1, /datum/reagent/consumable/nutriment/vitamin = 2)
 	list_reagents = list(/datum/reagent/consumable/nutriment = 8, /datum/reagent/consumable/nutriment/vitamin = 1)
 	bitesize = 1
-	w_class = WEIGHT_CLASS_NORMAL
```
Don't remove this.

Owai-Seek replied:
Forgot to restore that with the whitelisting.

Owai-Seek committed:
Fish Category, Misc Moves

Owai-Seek committed:
Gettin Salty. Fish is now Seafood.
